File: swift/1015_SmallestIntegerDivisibleByK.swift
class Solution {
    func smallestRepunitDivByK(_ K: Int) -> Int {
        if K % 2 == 0 || K % 5 == 0 { return -1 }
        
        var n = 1
        var length = 1
        while length < 10000000 {
            if n % K == 0 || n == 0 { return length }
            length += 1
            n = (n * 10 + 1) % K
        }
        return -1
    }
}

File: python/1015_SmallestIntegerDivisibleByK.py
class Solution:
    def smallestRepunitDivByK(self, K: int) -> int:
        if K % 2 == 0 or K % 5 == 0: return -1
        n = length = 1
        
        while n < 10000000:
            if n % K == 0 or n == 0: return length
            n = (n * 10 + 1) % K
            length += 1
        
        return -1
